app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
+  validates :name,  presence: true, length: {maximum: Settings.name_maxlength}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: Settings.email_maxlength},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: false}
```

Comment:
dòng này có tác dụng gì vậy?

Reply:
Dòng này dùng để valid email, presence: true (email not null), maximum (độ dài tối đa), format(định dạng của email phải là chữ cái, dấu rồi đến @ rồi đến chữ cái và dấu )